programs/storage_api/src/storage_instruction.rs

```diff
@@ -27,12 +40,71 @@ pub enum StorageInstruction {
     },
 }
 
-pub fn create_account(from: &Pubkey, to: &Pubkey, lamports: u64) -> Instruction {
-    system_instruction::create_account(&from, to, lamports, STORAGE_ACCOUNT_SPACE, &id())
+pub fn create_validator_storage_account(
+    from_pubkey: &Pubkey,
+    storage_pubkey: &Pubkey,
+    lamports: u64,
+) -> Vec<Instruction> {
+    vec![
+        system_instruction::create_account(
+            from_pubkey,
+            storage_pubkey,
+            lamports,
+            STORAGE_ACCOUNT_SPACE,
+            &id(),
+        ),
+        Instruction::new(
+            id(),
+            &StorageInstruction::InitializeValidatorStorage,
+            vec![AccountMeta::new(*storage_pubkey, false)],
+        ),
+    ]
 }
 
-pub fn mining_proof(
+pub fn create_replicator_storage_account(
```

Comment:
I'm less of a fan of this. It was cooler when it was implicitly figuring out what account to setup.

Reply:
Yeah it's cooler but much more difficult to reason about, which makes it easier to create something exploitable.   Programs should be simple and dumb, never sneaky.

Reply:
Well only certain instructions would implicitly setup the account. Others, like rewards, would return an error. Anyway what you have is totally fine. :)

Reply:
IMO that puts too much faith in how the program handles an uninitialized account, so easy to break those implicit assumptions and create an exploit.  I like code that's dumber than me.